app/app.js

```diff
@@ -0,0 +1,72 @@
+const express = require('express');
+const config = require("./config");
+const errors = require("./util/errors");
+const nunjucks = require("./nunjucks");
+const knex = require("./knex");
+const app = express();
+
+// shift.conf for nginx sets the x-forward header
+// and this says express can use it
+app.set('trust proxy', true);
+
+// allows ex. res.render('crawl.html');
+nunjucks.express(app);
+
+// modify every request
+app.use(function (req, res, next) {
+  // add these two error shortcuts:
+  res.textError = (msg) => errors.textError(res, msg);
+  res.fieldError = (fields, msg) => errors.fieldError(res, fields, msg);
+  // tbd: the php sets this for every end point.
+  // maybe unneeded with the trust_proxy call above?
+  res.set('Access-Control-Allow-Origin', "*");
```

Comment:
This should probably also be set to something less permissive. Not for this PR, but something to keep in mind.